### ISSUE TITLE
Proton 5.13-1 can't run games in Gnome Wayland session

### ISSUE THREAD

**Issue by NTMan** [ValveSoftware/Proton#4270]
Proton 5.13-1 can't run games in Gnome Wayland session, but in Gnome Xorg session games launched as intended.

Some logs with Wayland:
- [steam-1097150-Wayland.log](https://github.com/ValveSoftware/Proton/files/5389321/steam-1097150-Wayland.log)
- [steam-257420-Wayland.log](https://github.com/ValveSoftware/Proton/files/5389322/steam-257420-Wayland.log)
- [steam-614570-Wayland.log](https://github.com/ValveSoftware/Proton/files/5389323/steam-614570-Wayland.log)

the same games in Xorg:
- [steam-1097150-Xorg.zip](https://github.com/ValveSoftware/Proton/files/5389327/steam-1097150-Xorg.zip)
- [steam-257420-Xorg.log](https://github.com/ValveSoftware/Proton/files/5389332/steam-257420-Xorg.log)
- [steam-614570-Xorg.log](https://github.com/ValveSoftware/Proton/files/5389333/steam-614570-Xorg.log)

Previous Proton 5.0-10-rc4 launch games under both sessions well.

[Steam System Information report](https://gist.github.com/NTMan/33cca4906328793206218192820779ec)

**Comment by nutta-git** [comment]
#4266

**Comment by Oschowa** [comment]
I have the same issue, but found that launching the game through [gamescope](https://github.com/Plagman/gamescope) with e.g. `gamescope -- %command%` as launch option allows them to launch, even though gamescope is also technically a (x)wayland session.

**Comment by ghost** [comment]
Just confirming the issue, reverting back to 5.0-9 for the timebeing.

**Comment by Raklodder** [comment]
I'm on Fedora 33 gnome/wayland session, nothing works. Also tried running an Xorg session, but that didn't work either.

Quake Champions (2017), Prey (2006), Quake IV (2005), DOOM 3 (2004) or FarCry (2004).

I tried restarting my client, verify Proton and my installed games, run an Xorg session, but neither will launch.

Thankfully I can still manually fall back to proton 5.0-9 and the previous runtime.

Logfile: [vyfuMCwE.log](https://github.com/ValveSoftware/Proton/files/5392843/vyfuMCwE.log) Full system specifications: [9MRtQFwe.log](https://github.com/ValveSoftware/Proton/files/5392835/9MRtQFwe.log)

```
GameAction [AppID 611500, ActionID 2] : LaunchApp changed task to ProcessingInstallScript with ""
/data/src/clientdll/installscript_posix.cpp (357) : Assertion Failed: strCommonRedistForeignInstallPath != ""
/data/src/clientdll/installscript_posix.cpp (357) : Assertion Failed: strCommonRedistForeignInstallPath != ""
_v2-entry-point[137291]: STEAM_COMPAT_APP_ID=611500
_v2-entry-point[137291]: STEAM_COMPAT_SESSION_ID=87f8e6d8eb860e11
_v2-entry-point[137291]: XDG_RUNTIME_DIR=/run/user/1000
_v2-entry-point[137319]: STEAM_COMPAT_APP_ID=611500
_v2-entry-point[137319]: STEAM_COMPAT_SESSION_ID=87f8e6d8eb860e11
_v2-entry-point[137319]: XDG_RUNTIME_DIR=/run/user/1000
bwrap: Can't mkdir /usr/lib64/gconv: Read-only file system
ln: misslyckades att skapa symbolisk länk '/run/user/1000/SteamLinuxRuntime.87f8e6d8eb860e11/socket' → '': Filen eller katalogen finns inte
/data/src/clientdll/installscript_posix.cpp (419) : Assertion Failed: Standalone evaluator returned error code for app 611500
/data/src/clientdll/installscript_posix.cpp (419) : Assertion Failed: Standalone evaluator returned error code for app 611500
GameAction [AppID 611500, ActionID 2] : LaunchApp changed task to SiteLicenseSeatCheckout with ""
GameAction [AppID 611500, ActionID 2] : LaunchApp changed task to CreatingProcess with ""
GameAction [AppID 611500, ActionID 2] : LaunchApp waiting for user response to CreatingProcess ""
GameAction [AppID 611500, ActionID 2] : LaunchApp continues with user response "CreatingProcess"
Game update: AppID 611500 "", ProcID 137405, IP 0.0.0.0:0
ERROR: ld.so: object '/home/kattpojken/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/kattpojken/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/kattpojken/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
Starting app 611500
>>> Adding process 137405 for game ID 611500
GameAction [AppID 611500, ActionID 2] : LaunchApp changed task to WaitingGameWindow with ""
>>> Adding process 137406 for game ID 611500
ERROR: ld.so: object '/home/kattpojken/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
gamemodeauto: 
ERROR: ld.so: object '/home/kattpojken/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
gamemodeauto: 
pid 137432 != 137406, skipping destruction (fork without exec?)
ERROR: ld.so: object '/home/kattpojken/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
GameAction [AppID 611500, ActionID 2] : LaunchApp changed task to Completed with ""
_v2-entry-point[137406]: STEAM_COMPAT_APP_ID=611500
_v2-entry-point[137406]: STEAM_COMPAT_SESSION_ID=87f8e6d8eb860e11
_v2-entry-point[137406]: XDG_RUNTIME_DIR=/run/user/1000
ERROR: ld.so: object '/home/kattpojken/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
>>> Adding process 137424 for game ID 611500
>>> Adding process 137433 for game ID 611500
>>> Adding process 137436 for game ID 611500
>>> Adding process 137441 for game ID 611500
bwrap: Can't mkdir /usr/lib64/gconv: Read-only file system
ln: misslyckades att skapa symbolisk länk '/run/user/1000/SteamLinuxRuntime.87f8e6d8eb860e11/socket' → '': Filen eller katalogen finns inte
ERROR: ld.so: object '/home/kattpojken/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
>>> Adding process 137475 for game ID 611500
bwrap: Can't mkdir /usr/lib64/gconv: Read-only file system
ln: failed to create symbolic link '/run/user/1000/SteamLinuxRuntime.87f8e6d8eb860e11/socket' -> '': No such file or directory
ERROR: ld.so: object '/home/kattpojken/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
ERROR: ld.so: object '/home/kattpojken/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
pressure-vessel-launch[137556]: Can't connect to peer socket: Could not connect: No such file or directory
>>> Adding process 137555 for game ID 611500
>>> Adding process 137556 for game ID 611500
Game removed: AppID 611500 "", ProcID 137405 
Uploaded AppInterfaceStats to Steam
Exiting app 611500
```

**Comment by D-357** [comment]
Wayland is fine for me on Fedora 32. Here's my log for anyone to compare
[steam-976730-wayland.log](https://github.com/ValveSoftware/Proton/files/5392654/steam-976730.log)

**Comment by kermeat** [comment]
Works on Fedora 33 gnome/wayland if soldier runtime disabled

**Comment by Mershl** [comment]
@kermeat Facing the same issue on F33 gnome+wayland. How did you disable the soldier runtime?

**Comment by kermeat** [comment]
@Mershl I edited `<steam-library>/steamapps/common/SteamLinuxRuntime_soldier/_v2-entry-point`
https://github.com/flightlessmango/MangoHud/issues/369#issuecomment-709902078

**Comment by Mershl** [comment]
I'm seeing the following exception when trying to run games (tested Hades) on Proton5.13-1 on Gnome Wayland:
```
[00000000000000CC:] EXCEPTION handling: System.ComponentModel.Win32Exception: Error creating window handle.

Unhandled Exception:
System.ComponentModel.Win32Exception
 (0x80004005): Error creating window handle.
```

Full log: [steam-1145360.log](https://github.com/ValveSoftware/Proton/files/5396061/steam-1145360.log)

**Comment by Oschowa** [comment]
This mutter MR makes games work in a gnome wayland session when using the new runtime: https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1508

**Comment by smcv** [comment]
> This mutter MR makes games work in a gnome wayland session when using the new runtime

I think it'd be better to fix this in pressure-vessel. We believe we have a fix for this.

Workaround: use an X11 session for now.

> if soldier runtime disabled

Please don't do that, it isn't supported or supportable.

**Comment by smcv** [comment]
@Raklodder wrote:
> Also tried running an Xorg session, but that didn't work

This means the root cause is not the same for you as for the original reporter of this issue. Please open a separate issue mentioning `bwrap: Can't mkdir /usr/lib64/gconv: Read-only file system` in the title, if there isn't already one.

**Comment by smcv** [comment]
> Wayland is fine for me on Fedora 32

This is because the GNOME Shell change that triggers this is new in Fedora 33.

**Comment by ghost** [comment]
> I have the same issue, but found that launching the game through [gamescope](https://github.com/Plagman/gamescope) with e.g. `gamescope -- %command%` as launch option allows them to launch, even though gamescope is also technically a (x)wayland session.

Gamescope doesn't work for me, somehow it is not used when switching from Proton 5.0-9 to 5.13-1.

**Comment by smcv** [comment]
Please try again with the beta branch of both the Steam client and "Steam Linux Runtime - soldier".

If you were previously getting X11 or XAUTHORITY errors similar to ValveSoftware/steam-runtime#283 or #4266, those should be resolved in the beta version.

If errors similar to `bwrap: Can't mkdir /usr/lib64/gconv: Read-only file system` persist with that version, please open a separate issue, mentioning the error message from `bwrap` in the issue title. We think this is a different bug, unrelated to X11 and XAUTHORITY.

**Comment by Raklodder** [comment]
~~I can confirm that the latest steam beta https://github.com/ValveSoftware/Proton/issues/4289#issuecomment-715346209 solved my gnome/wayland issue on Fedora 33 after a clean install. All I did was to download and install steam, changed to beta updates, went to my settings and enabled steam play for all other titles and begun to download Quake Champions (it automatically chose to download proton 5.13-1) and when I started to install my game it also chose to install the steam linux soldier runtime, when that was done and all of my shaders were built it just worked.~~

@kisak-valve looks like i spoke too soon, a few hours of qc gameplay, a system restart and now nothing works, same as before and i'm back to using proton 5.0-9, mind you that i didn't install it prior to 5.13-1 or the soldier runtime.

```
GameAction [AppID 611500, ActionID 1] : LaunchApp changed task to ProcessingInstallScript with ""
_v2-entry-point[87605]: STEAM_COMPAT_APP_ID=611500
_v2-entry-point[87605]: STEAM_COMPAT_SESSION_ID=e2ca29bdd0a8a370
_v2-entry-point[87605]: XDG_RUNTIME_DIR=/run/user/1000
_v2-entry-point[87629]: STEAM_COMPAT_APP_ID=611500
_v2-entry-point[87629]: STEAM_COMPAT_SESSION_ID=e2ca29bdd0a8a370
_v2-entry-point[87629]: XDG_RUNTIME_DIR=/run/user/1000
/usr/lib/x86_64-linux-gnu/gio/modules/libdconfsettings.so: undefined symbol: g_type_ensure
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libdconfsettings.so
/usr/lib/x86_64-linux-gnu/gio/modules/libdconfsettings.so: undefined symbol: g_type_ensure
Failed to load module: /usr/lib/x86_64-linux-gnu/gio/modules/libdconfsettings.so
esync: up and running.
_v2-entry-point[87764]: STEAM_COMPAT_APP_ID=611500
_v2-entry-point[87764]: STEAM_COMPAT_SESSION_ID=e2ca29bdd0a8a370
_v2-entry-point[87764]: XDG_RUNTIME_DIR=/run/user/1000
_v2-entry-point[87784]: STEAM_COMPAT_APP_ID=611500
_v2-entry-point[87784]: STEAM_COMPAT_SESSION_ID=e2ca29bdd0a8a370
_v2-entry-point[87784]: XDG_RUNTIME_DIR=/run/user/1000
_v2-entry-point[87804]: STEAM_COMPAT_APP_ID=611500
_v2-entry-point[87804]: STEAM_COMPAT_SESSION_ID=e2ca29bdd0a8a370
_v2-entry-point[87804]: XDG_RUNTIME_DIR=/run/user/1000
GameAction [AppID 611500, ActionID 1] : LaunchApp changed task to SiteLicenseSeatCheckout with ""
GameAction [AppID 611500, ActionID 1] : LaunchApp changed task to CreatingProcess with ""
GameAction [AppID 611500, ActionID 1] : LaunchApp waiting for user response to CreatingProcess ""
GameAction [AppID 611500, ActionID 1] : LaunchApp continues with user response "CreatingProcess"
Opted-in Controller Mask for AppId 611500: 0
Game update: AppID 611500 "", ProcID 87833, IP 0.0.0.0:0
ERROR: ld.so: object '/home/kattpojken/.local/share/Steam/ubuntu12_32/gameoverlayrenderer.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
Starting app 611500
>>> Adding process 87833 for game ID 611500
GameAction [AppID 611500, ActionID 1] : LaunchApp changed task to WaitingGameWindow with ""
/bin/sh: PROTON_NO_ESYNC: command not found
GameAction [AppID 611500, ActionID 1] : LaunchApp changed task to Completed with ""
>>> Adding process 87834 for game ID 611500
Game removed: AppID 611500 "", ProcID 87833 
Uploaded AppInterfaceStats to Steam
Exiting app 611500
```

**Comment by kisak-valve** [comment]
Hello @Raklodder, `/bin/sh: PROTON_NO_ESYNC: command not found` looks anomalous and different from the issue being tracked here. Please open a new issue report after checking the game's launch options.

**Comment by Raklodder** [comment]
My apologies, I'd forgot to remove the "PROTON_NO_ESYNC" command from my launch options, though I get the same result without it. However I won't spam until a new proton build and/or the mentioned gnome/mutter fix has been released, but just to let you know, when I remove the launch option, it will say "esync is up and running", but unfortunately that's were the fun ends. In other words, the bug report I made on october 16th above remains, but again reverting back to 5.0-9 is sufficient for now.

**Comment by kisak-valve** [comment]
Closing as fixed in Gnome 3.38.2 by https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1508.

**Comment by smcv** [comment]
@Raklodder wrote:
> the bug report I made on october 16th above remains

If that's still true, please report it *separately* to <https://github.com/ValveSoftware/steam-runtime/issues>, mentioning `bwrap: Can't mkdir /usr/lib64/gconv: Read-only file system` in the title. It looks like a separate issue, and if we have several issues mixed up in one issue report, managing the issue reports takes up time that we should be spending on fixing bugs.

A log with `PRESSURE_VESSEL_VERBOSE=1` in the environment would be useful information - again, please send that to a new bug report, not this one. You can do this without the extra complexity of involving the Steam client and Proton with a command like this:

```
cd /path/to/SteamLinuxRuntime_soldier
PRESSURE_VESSEL_VERBOSE=1 ./run -- steam-runtime-system-info --verbose 2>&1 | tee container.log
```

and include container.log with the new bug report. You can edit/censor the log if there's anything in it that you consider private, as long as it's obvious where it has been edited, for instance replacing your username with REDACTED.
